fem/tmop_tools.cpp

```diff
@@ -33,10 +33,11 @@ void AdvectorCG::ComputeAtNewPosition(const Vector &new_nodes,
    const int pnt_cnt = new_field.Size()/ncomp;
 
    new_field = field0;
-
+   new_field.HostReadWrite();
```

Comment:
This transfer to host seems unnecessary here.

Reply:
even with the following `new_field_temp.MakeRef`?

Reply:
maybe, the Min/Max could be done on the device, not sure about the `ode_solver.Step`, but there is a `HostReadWrite` before the host over/under shoots, I agree.

Reply:
`MakeRef` just needs the base to be registered (at least until #2203 or similar change is merged) but there is no need for it to be on host -- it must work regardless of where the data is.

Reply:
looks good w/o it - I'll remove it

Reply:
`done`